tls/s2n_tls13_secrets.c

```diff
@@ -671,3 +694,53 @@ S2N_RESULT s2n_tls13_secrets_get(struct s2n_connection *conn, s2n_extract_secret
     RESULT_ENSURE_GT(secret->size, 0);
     return S2N_RESULT_OK;
 }
+
+int s2n_connection_tls_exporter(
+        struct s2n_connection *conn,
+        uint8_t *output_,
+        uint32_t output_length,
+        const uint8_t *label_,
+        uint32_t label_length,
+        const uint8_t *context,
+        uint32_t context_length)
+{
+    POSIX_ENSURE_REF(conn);
+    POSIX_ENSURE_REF(output_);
+    POSIX_ENSURE_REF(label_);
+    POSIX_ENSURE_REF(context);
+    POSIX_ENSURE(s2n_connection_get_protocol_version(conn) == S2N_TLS13, S2N_ERR_INVALID_STATE);
+
+    s2n_hmac_algorithm hmac_alg = conn->secure->cipher_suite->prf_alg;
```

Comment:
This is less safe than you'd expect. If this is called during the handshake, it could seg fault. You can use [is_handshake_complete](https://github.com/aws/s2n-tls/blob/main/tls/s2n_connection.h#L50) to explicitly check that the handshake is done, but it's also probably a good idea to just sanity check with some ENSURE_REFs.

```suggestion

    POSIX_ENSURE(is_handshake_complete, S2N_ERR_HANDSHAKE_NOT_COMPLETE);
    POSIX_ENSURE_REF(conn->secure);
    POSIX_ENSURE_REF(conn->secure->cipher_suite);
    s2n_hmac_algorithm hmac_alg = conn->secure->cipher_suite->prf_alg;
```